### ISSUE TITLE
Missing methods to support scala-java-time

### ISSUE THREAD

**Issue by cquiroz** [scala-native/scala-native#1023]
Trying to port scala-java-time and run an example I found the following missing methods:

```
[error] cannot link: @java.util.List::contains_java.lang.Object_bool
[error] cannot link: @java.util.List::toArray_scala.scalanative.runtime.ObjectArray_scala.scalanative.runtime.ObjectArray
[error] cannot link: @java.util.Map::putIfAbsent_java.lang.Object_java.lang.Object_java.lang.Object
[error] cannot link: @java.util.concurrent.ConcurrentHashMap
[error] cannot link: @java.util.concurrent.ConcurrentHashMap::init
[error] cannot link: @java.util.concurrent.ConcurrentHashMap::init_i32_f32_i32
[error] cannot link: @java.util.concurrent.ConcurrentMap
[error] cannot link: @java.util.concurrent.CopyOnWriteArrayList
[error] cannot link: @java.util.concurrent.CopyOnWriteArrayList::init
```

**Comment by densh** [comment]
Thanks for the report, those seem to be easy to fix.

**Comment by Andrei-Pozolotin** [comment]
+1

**Comment by LeeTibbert** [comment]
An update: 

After considerable, for me, sandbox work I have put this work on pause.

An implementation likely to pass review would use  java default methods in traits. This is implemented
in scala 2.12, but not in the scala 2.11 currently supported by SN.  I proved to myself that required
cleanup and implementation __could__ be done with current SN, but would introduce too much
technical debt. The dog could sing, but not well, and would need to be retrained soon.

**Comment by LeeTibbert** [comment]
TL; DR - Progress but not completion

\scala-java-time has been broken up into modular sub-projects since the base Issue was created 3+ years ago.
One of the current projects is also named scala-java-time.  There are other projects for the timezone database
and locale support.

As of PR #10910, and the earlier PRs in that series, a number of scala-java-* projects at least compile for 
SN 0.4.0-SNAPSHOT-Post-Aug2020.  I am not sure if I got them all, but I got the ones I needed in order to get
 scala-java-time compiling, linking, and executing its tests in my local sandbox.  
That sandbox contains changes to scala-java-* once SN 0.Next and portable-scala.reflect for itare released.   
Scaffolding to support the corresponding version of ScalaTest is also required (the fix is no-evident, at least
to me).

The results of running the scala-java-time  ensemble look like a Christmas tree; red (failures) & green (successes).
I did not examine each and every failure case. The ones did scan were caused by unknown timezone Ids.  I believe
those are valid failures because I built my sandbox as minimal to support my debugging/development.

I am not closing this Issue yet or even when the PRs in this series have been merged
At some time in the future, it would be nice to see green across the board.  Perhaps by the New Year, leave
the lights up for this Christmas.

I am not closing this Issue yet, or even after the series has been merged because I would like to
exercise the scala-java-* ecosystem more thoroughly before declaring victory.

**Comment by LeeTibbert** [comment]
Truth in advertising: I have done _absolutely_ no_ performance work on any of the above. I was/am
focusing on first existence and then correctness.

**Comment by ekrich** [comment]
@cquiroz Now that `Locale` and `java.time` and friends are removed via #2087 and #2079, do we have the methods needed? @LeeTibbert Do you know?

**Comment by LeeTibbert** [comment]
@ekrich , @WojciechMazur Last time I did a port of scala-java-time, about 3 months or so ago,  IIRC, sjt itself 
built and worked well enough for ZIO.  The sticking point was that the s-j-t tests did not build.  I believe
that once I took the recursive edit to get scalatest working with the Fake-M2 I was using, that the s-j-t tests
worked.  Then again, I might be in wish fulfillment mode.

I have implemented these methods four or five times now,  but they never saw a merge.  There
is an old PR, probably referencing this one, which probably has an implementation.

I think the point of action here is a trial build of scala-java-time with a pre-publication
version of 0.4.0.   Sorry, but events in Real Life prevent me from stepping in and
doing that just now.

**Comment by ekrich** [comment]
@LeeTibbert Thanks for the info. We can wait on @cquiroz to see if this can be closed or can be updated.

**Comment by LeeTibbert** [comment]
My memory is that the more difficult 0.4.0-mumble changes, post Summer 2020, to s-j-t were
to the test-runner.  That is, as mentioned, s-j-t core compiled and ran fine, but that the
s-j-t SN-Test environment (snd scalatest) needed work.  The former would probably go
a lot easier for a SN contributor.

I have all of the s-j-t changes required as of Fall 2020 queued in my private sandbox
After SN 0.4.0 is released  and if Life is kind to me, I will try to create/find the 
cycles to see if we, the SN community, can submit a series of PRs to s-j-t.

Now that SN is recommending 3rd party java.time, it would be nice to have
at least one that works reasonable (I believe that @ekrich has some work
on another implementation.  The thing that recomments s-j-t to my mind
is it extensive test set (and timezone DB support).)

I try to console myself that everything can not be done in one day...
At least the SN eco system is now making rapid progress..

**Comment by lolgab** [comment]
This can be closed now that scala-java-time for Scala Native is merged https://github.com/cquiroz/scala-java-time/pull/346
